manifests/config/apache.pp

```diff
@@ -0,0 +1,284 @@
+  Integer[0] $passenger_start_timeout = $::foreman::passenger_start_timeout,
+  Stdlib::HTTPUrl $foreman_url = $::foreman::foreman_url,
+  Boolean $keepalive = $::foreman::keepalive,
+  Integer[0] $max_keepalive_requests = $::foreman::max_keepalive_requests,
+  Integer[0] $keepalive_timeout = $::foreman::keepalive_timeout,
+  Optional[String] $access_log_format = undef,
+  Boolean $ipa_authentication = $::foreman::ipa_authentication,
+  Optional[Boolean] $selinux = $::foreman::selinux,
+) {
+  $docroot = "${app_root}/public"
+  $suburi_parts = split($foreman_url, '/')
+  $suburi_parts_count = size($suburi_parts) - 1
+  if $suburi_parts_count >= 3 {
+    $suburi_without_slash = join(values_at($suburi_parts, ["3-${suburi_parts_count}"]), '/')
+    if $suburi_without_slash {
+      $suburi = "/${suburi_without_slash}"
+    } else {
+      $suburi = undef
+    }
+  } else {
+    $suburi = undef
+  }
+
+  if $passenger {
+    $passenger_options = {
+      'passenger_app_root' => $app_root,
+      'passenger_min_instances' => $passenger_min_instances,
+      'passenger_start_timeout' => $passenger_start_timeout,
+      'passenger_ruby' => $passenger_ruby,
+    }
+    $passenger_http_prestart = $passenger_prestart ? {
+      true  => "http://${servername}:${server_port}",
+      false => undef,
+    }
+    $passenger_https_prestart = $passenger_prestart ? {
+      true  => "https://${servername}:${server_ssl_port}",
+      false => undef,
+    }
+
+    if $suburi {
+      $custom_fragment = template('foreman/_suburi.conf.erb')
+    } else {
+      $custom_fragment = template('foreman/_assets.conf.erb')
+    }
+
+    $proxy_http_options = {}
+    $proxy_https_options = {}
+
+    if $app_root {
+      file { ["${app_root}/config.ru", "${app_root}/config/environment.rb"]:
+        owner => $user,
+      }
+    }
+  } else {
+    $passenger_options = {}
+    $passenger_http_prestart = undef
+    $passenger_https_prestart = undef
+
+    if $suburi {
+      $custom_fragment = undef
+    } else {
+      $custom_fragment = template('foreman/_assets.conf.erb')
+    }
+
+    $backend = 'http://localhost:3000/'
+
+    $proxy_http_options = {
+      'proxy_preserve_host' => true,
+      'proxy_add_headers'   => true,
+      'request_headers' => [
+        'set X_FORWARDED_PROTO "http"',
+      ],
+      'proxy_pass'          => {
+        'no_proxy_uris' => ['/pulp', '/streamer', '/pub'],
+        'path'          => '/',
+        'url'           => $backend,
+        'params'        => {'retry' => '0'},
+      },
+    }
+
+    $proxy_https_options = {
+      'ssl_proxyengine' => true,
+      'request_headers' => [
+        'set X_FORWARDED_PROTO "https"',
+        'set SSL_CLIENT_S_DN "%{SSL_CLIENT_S_DN}s"',
+        # Broken - needs https://github.com/theforeman/foreman/pull/6239
+        'set SSL_CLIENT_CERT "%{SSL_CLIENT_CERT}s"',
+        'set SSL_CLIENT_VERIFY "%{SSL_CLIENT_VERIFY}s"',
```

Comment:
I thought you wanted to unset these and then set them for a given location?

Reply:
I need to verify that